blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,262 @@
+            return id;
+        }
+
+        @Override
+        public AbstractFolderProperty<?> reconfigure(StaplerRequest req, JSONObject form) throws FormException {
+            return this;
+        }
+
+        @Extension(optional = true)
+        public static class DescriptorImpl extends AbstractFolderPropertyDescriptor {
+        }
+
+        public static class CredentialsStoreActionImpl extends CredentialsStoreAction {
+
+            private final CredentialsStore store;
+
+            CredentialsStoreActionImpl(CredentialsStore store) {
+                this.store = store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Nonnull
+            @Override
+            public CredentialsStore getStore() {
+                return store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconFileName() {
+                return isVisible()
+                        ? "/plugin/credentials/images/48x48/folder-store.png"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconClassName() {
+                return isVisible()
+                        ? "icon-credentials-folder-store"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return DISPLAY_NAME;
+            }
+
+            @Override
+            public String getUrlName() {
+                return super.getUrlName();
+            }
+        }
+
+        public static class StoreImpl extends CredentialsStore{
+            private final AbstractFolder abstractFolder;
+            private final CredentialsStoreAction storeAction;
+
+            StoreImpl(AbstractFolder abstractFolder) {
+                super(BlueOceanCredentialsProvider.class);
+                this.abstractFolder = abstractFolder;
+                this.storeAction = new CredentialsStoreActionImpl(this);
+            }
+
+            @Nonnull
+            @Override
+            public List<Domain> getDomains() {
+                //XXX: how to get domain from given Credentials object? Maybe attach domain name to AbstractFolder?
+                return Collections.emptyList();
+            }
+
+            @Nullable
+            @Override
+            public CredentialsStoreAction getStoreAction() {
+                return storeAction;
+            }
+
+            @Nonnull
+            @Override
+            public ModelObject getContext() {
+                return abstractFolder;
+            }
+
+            @Override
+            public boolean hasPermission(@Nonnull Authentication a, @Nonnull Permission permission) {
+                return abstractFolder.getACL().hasPermission(a,permission);
```

Comment:
Do need to tweak this, as this is a read only domain we want to always return false for the `Credentials/CREATE`, `Credentials/DELETE`, `Credentials/MANAGE_DOMAIN` and `Credentials/UPDATE` permissions